packages/fxa-react/components/Modal/index.tsx

```diff
@@ -3,60 +3,79 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import React, { ReactNode } from 'react';
-import { useClickOutsideEffect } from 'fxa-react/lib/hooks';
-import { useEscKeydownEffect, useChangeFocusEffect } from '../../lib/hooks';
+import { useClickOutsideEffect, useEscKeydownEffect, useChangeFocusEffect } from 'fxa-react/lib/hooks';
 import classNames from 'classnames';
 import Portal from 'fxa-react/components/Portal';
 import { ReactComponent as CloseIcon } from 'fxa-react/images/close.svg';
 import { Link, useLocation } from '@reach/router';
 import { Localized, useLocalization } from '@fluent/react';
 
+import './index.scss';
+
 type ModalProps = {
+  id: string,
   className?: string;
-  onDismiss: () => void;
+  onDismiss?: () => void;
   onConfirm?: () => void;
   children: ReactNode;
   hasButtons?: boolean;
   hasCancelButton?: boolean;
-  headerId: string;
-  descId: string;
+  headerId?: string;
+  descId?: string;
   route?: string;
   confirmText?: string;
   confirmBtnClassName?: string;
   'data-testid'?: string;
 };
 
+/* istanbul ignore next - not worth testing this function */
+const noop = () => {};
+
 export const Modal = ({
-  className = '',
+  id,
+  className,
   onDismiss,
   onConfirm,
   children,
-  hasButtons = true,
-  hasCancelButton = true,
+  hasButtons,
+  hasCancelButton,
   headerId,
   descId,
   route,
   confirmText = 'Confirm',
   confirmBtnClassName = 'cta-primary',
-  'data-testid': testid = 'modal',
+  'data-testid': testId,
 }: ModalProps) => {
-  const modalInsideRef = useClickOutsideEffect<HTMLDivElement>(onDismiss);
+  const modalInsideRef = useClickOutsideEffect<HTMLDivElement>(onDismiss || noop);
```

Comment:
The type assigned to `onDismiss` in the DialogMessage component was `Function` and therefore included `noop` in case it was empty/ `() => {}`, while the type assigned to `onDismiss` in the previous Modal component was `() => void`. I'll see if I can update them all to be consistent with `() => void`.

Reply:
Ah, it stems from `locationReload` being set to `() => {}` in `defaultAppContext` from `fxa-payments-server/src/lib/AppContext.tsx`. 🤔